source/opt/const_folding_rules.cpp

```diff
@@ -196,7 +254,84 @@ double GetDoubleFromConst(const analysis::Constant* c) {
   }
 }
 
-// This macro defines a |FloatScalarFoldingRule| that applies |op|.  The
+// Returns the integer value of |c|.  The constant |c| must have type
+// |Integer|, and width |32|.
+uint32_t GetIntegerFromConst(const analysis::Constant* c) {
+  assert(c->type()->AsInteger() != nullptr &&
+         c->type()->AsInteger()->width() == 32);
+  const analysis::IntConstant* ic = c->AsIntConstant();
+  if (ic) {
+    return ic->GetU32BitValue();
+  } else {
+    assert(c->AsNullConstant() && "c must be an integer constant.");
+    return 0.0f;
+  }
+}
+
+// This macro defines a |UnaryScalarFoldingRule| that performs float to
+// integer conversion.
+// TODO(greg-lunarg): Support for 64-bit integer types.
+#define FOLD_FTOI_OP()                                                        \
```

Comment:
I used a macro for the floating point binary operations because I could not find a way to pass in the operator.  In this case there is not parameter to the macro.  If there is no good reason for it, I would prefer to not use a macro.

Reply:
Gone.